### DISCUSSION TITLE
使用WS时，API请求和返回可增加字段caller_info

### DISCUSSION THREAD

**Discussion by awp9835** [howmanybots/onebot#146] [想法 / 建议]
最近自己也在造轮子，把多个第三方插件整合在一个新的插件里，并且添加一些功能和钩子。由于使用消息队列和onebot通信，接收API的调用结果，所以无法得知API的调用者是哪个插件。
建议在API请求和返回时添加一个caller_info字段，用来识别调用者信息。该字段在请求时由调用方添加；如果该字段存在，onebot原样在返回的结果中原样保留该字段。

**Comment by undefined-moe** [comment]
emm有个东西叫echo字段（）

**Comment by awp9835** [comment]
> emm有个东西叫echo字段（）
标准里没有写这个字段

**Comment by awp9835** [comment]
> emm有个东西叫echo字段（）

这个字段的说明在communication里，应该在API概述里加一条链接指向该内容
